apps/maestro/src/features/SendInterchainToken/hooks/useInterchainTokenServiceTransferMutation.ts

```diff
@@ -48,9 +52,24 @@ export function useInterchainTokenServiceTransferMutation(
 
   const { address } = useAccount();
 
+  const { data: tokenManagerAddress } =
+    useReadInterchainTokenServiceTokenManagerAddress({
+      args: INTERCHAIN_TOKEN_SERVICE_ENCODERS.tokenManagerAddress.args({
+        tokenId: config.tokenId,
+      }),
+      query: {
+        enabled: chainId === HEDERA_CHAIN_ID && Boolean(config.tokenId),
```

Comment:
What about using an array here too, for the chains which need this functionality?